File: TEAM B2 (Season 2)/Week 2/Musk/[스택_큐_프로세스].cpp
#include <string>
#include <vector>
#include <queue>
#include <iostream>
#include <algorithm> 
using namespace std;

int solution(vector<int> priorities, int location) {
    int answer = 0;
    int count = 0;
    queue<int> q;
    queue<int> index;
    for(int i=0;i<priorities.size();i++) { 
        q.push(priorities[i]);
        index.push(i);
    }
    
    while(!q.empty()) {
        int max = *max_element(priorities.begin(), priorities.end());
        
        int temp = q.front();
        q.pop();
        int index_temp = index.front();
        index.pop();
        
        if(max == temp) {
            count++;
            vector<int>::iterator maxElementIterator = max_element(priorities.begin(), priorities.end());
             priorities.erase(maxElementIterator);
            if(index_temp == location) {
                answer = count;
                break;
            }
        }
        else {
            q.push(temp);
            index.push(index_temp);
        }
        
    }
    return answer;
}


        // int temp = priorities.front();
        // priorities.erase(priorities.begin());
        // cout << temp;

File: TEAM B2 (Season 2)/Week 2/Moana/[스택큐_프로세스].py
from collections import deque

def solution(priorities, location):
    answer = 1
    ind = [0 for _ in range(len(priorities))]
    ind[location] = 1
    index = deque(ind)
    priority = deque(priorities)
    
    while True:
        tmp = max(priority)
        if priority[0] != tmp:
            priority.append(priority.popleft())
            index.append(index.popleft())
        elif priority[0] == tmp:
            if index[0] == 1:
                break
            else:
                priority.popleft()
                index.popleft()
                answer += 1

    return answer

